optuna/_gp/gp.py

```diff
@@ -164,7 +166,8 @@ def _fit_kernel_params(
             np.log(initial_kernel_params.inverse_squared_lengthscales.detach().numpy()),
             [
                 np.log(initial_kernel_params.kernel_scale.item()),
-                np.log(initial_kernel_params.noise_var.item() - minimum_noise),
+                # We add 0.01 * minimum_noise to initial noise_var to avoid instability.
+                np.log(initial_kernel_params.noise_var.item() - 0.99 * minimum_noise),
```

Comment:
Based on the comment above (i.e., "add 0.01 * minimum_noise to initial noise_var"), I feel this line should be like below.
```
np.log(initial_kernel_params.noise_var.item() + 0.01 * minimum_noise)
```

Reply:
I think this line is trying to transform the variable so that the search space is gonna be $(-\infty, \infty)$.

Reply:
Memo: The intention is below.
```
(initial_noise_var + 0.01 * minimum_noise) - minimum_noise = initial_var - 0.99 * minimum_noise.
```